### PR TITLE
feat: [PL-24534]: Add new git aware persistence interface for gitX Simplification

### PR DESCRIPTION
**Pull request by mohitg0795**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32309)

### CODE REVIEW

File: 800-pipeline-service/src/main/java/io/harness/pms/pipeline/PipelineEntity.java [hunk 1]
```diff
@@ -170,4 +172,29 @@ public String getInvalidYamlString() {
   public boolean shouldAllowStageExecutions() {
     return allowStageExecutions != null && allowStageExecutions;
   }
+
+  @Override
```
Comment (namanvermah) [inline 800-pipeline-service/src/main/java/io/harness/pms/pipeline/PipelineEntity.java:176]: These get methods are already handled by lombok

**Review by namanvermah — DISMISSED** [review]
:lgtm:

> Reviewed 9 of 16 files at r2, 2 of 2 files at r3, 5 of 5 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32309#-:-N05Z9Y24sOBVcfU1Jqr:bnfp4nl)* status: 16 file(s) to review. 1 of 2 approvals obtained.

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 5 of 16 files at r2, 1 of 2 files at r3, 4 of 5 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32309)* status: 10 file(s) to review. 1 of 2 approvals obtained.

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 5 of 16 files at r2, 1 of 2 files at r3, 2 of 5 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32309)* status: 7 file(s) to review.

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 1 of 16 files at r2, 4 of 5 files at r4, 2 of 5 files at r5.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32309)* status: :shipit: complete! ✓

**Pull request merged**